aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUser.java

```diff
@@ -952,6 +952,7 @@ protected CognitoUserSession getCachedSession() {
                     clearCachedTokens();
                     throw new CognitoNotAuthorizedException("User is not authenticated", nae);
                 } catch (final Exception e) {
+                    clearCachedTokens();
```

Comment:
What scenario are you hitting that requires the tokens to be cleared?

Reply:
I can't find the cause, but user information is not deleted from cache after logging out. (I checked the shared preferences file and checked). The same user cannot login again. The problem is correcting after cache is deleted. We are using user pool without using the Identity pool. I didn't find out how to delete Cache.

Reply:
Class: Cognito User /  line :2323 

cognitoIdentityProviderClient .initiateAuth(initiateAuthRequest)
returns
'com.amazonaws.services.cognitoidentityprovider.model.UserNotFoundException' exception.

<img width="1245" alt="Screen Shot 2019-04-09 at 10 27 36" src="https://user-images.githubusercontent.com/10996313/55781176-3ddbbc00-5ab2-11e9-9701-0b1552385c2b.png">

Reply:
https://github.com/aws-amplify/aws-sdk-android/issues/231

Reply:
Can you change your edit by creating another catch statement specifically for UserNotFoundException? This is because network not being available also manifests in an exception and we want to keep the session in that case.

Can you share which keys are not being deleted?

Reply:
I think I've found that why user information remains in shared preferences.

In app **AndroidManifest.xml**
`android:allowBackup="true"`

If user deletes the app while logged in, user's information remains in shared preferences file (CognitoIdendityProviderCache). If user reinstall  after Session has dropped, it cannot log in with this information.  Still I don't know why throw UserNotFoundException. 

I changed manifest file; 

`android:allowBackup="false"`
and
`tools:replace="android:allowBackup"`